### PR TITLE
fix(builders): :bug: matching special characters and numbers in commit scope

### PR DESCRIPTION
**Pull request by wSedlacek**
## Description

This fixes projects from being skipped when using `-` in their names
This fixes projects from being skipped when using numbers in their names

closes: #177

### CODE REVIEW

File: libs/builders/src/semantic-release/lib/get-analyze-commits-options.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ export function getAnalyzeCommitsOptions(project: string, mode: 'independent' |
     parserOpts = { headerPattern: new RegExp(`^(\\w*)(?:\\((${project}|\\*|deps)\\))?: (.*)$`) };
     releaseRules = releaseRules.map((rule) => ({ ...rule, scope: project }));
   } else {
-    parserOpts = { headerPattern: new RegExp(`^(\\w*)(?:\\((\\w*)\\))?: (.*)$`) };
+    parserOpts = { headerPattern: new RegExp(`^(.*)(?:\\((\\w*)\\))?: (.*)$`) };
```
Comment (samuelfernandez) [inline libs/builders/src/semantic-release/lib/get-analyze-commits-options.ts:14]: The first part of the regex is for the commit type, such as `feat`, `fix` and so on, and will always be a word. Actually the fix should be in the scope part:

```suggestion
    parserOpts = { headerPattern: new RegExp(`^(\\w*)(?:\\(([^:]*)\\))?: (.*)$`) };
```

You can test it here https://regex101.com/r/efCmKF/1/

**Comment by samuelfernandez** [comment]
Thanks a lot @wSedlacek for catching this bug, and for your first contrib to the project! 🚀 Gonna merge and release a new version

Besides that, I'd be curious to know if you are using the sync mode in any open source project in GitHub. I'm planning to make some enhancements on that, maybe providing another builder to run all releasable projects in just one command. Any feedback on better ergonomics, or other bugs, would be much appreciated

File: libs/builders/src/semantic-release/lib/get-analyze-commits-options.ts (hunk 1 above)
Comment (wSedlacek) [reply libs/builders/src/semantic-release/lib/get-analyze-commits-options.ts]: Good catch. I shouldn't have written that at 1 AM.

**Comment by wSedlacek** [comment]
@samuelfernandez,
> I'd be curious to know if you are using the sync mode in any open source project in GitHub

I have been doing some research for work. I hope that we will be using this project for our releases in the near future for both our open source and proprietary npm projects. We have a lot on our plate, so it might be quite a bit before we actually get to that point. From my research thus far this project has the most promise.

> maybe providing another builder to run all releasable projects in just one command

This would be very useful.
I noticed that there is a `tag` for each project when doing the releases one by one.
It would be nice if there was just a single `tag` and release commit.
I also noticed that the `package.json` wouldn't be updated for projects that are not configured to release with this builder.
There probably are few cases where apps or libs wouldn't be configured to update, but just in case it might be nice to bump the version for all projects just so they stay in sync.

> Any feedback on better ergonomics, or other bugs, would be much appreciated

It would be nice if the `branches` option was configurable. That setting appears to be quite project specific.

It might be nice to configure the release commit, as it was written I had to change some of my commit lint rules to abide by it's format, ie add `release` as a valid scope and remove the body requirement.

Perhaps bumping the root `package.json` version would be useful?

I did run into some problems configuring an admin git creds with the example script, the solution was to set env variables and update the `origin` URL. Perhaps that could be documented?

```yml
      - name: Fetch latest base branch
        env:
          GIT_USERNAME: ${{ secrets.GIT_USERNAME }}
          GIT_TOKEN: ${{ secrets.RELEASE_TOKEN }}
        run: |
          git remote set-url origin "https://${GIT_USERNAME}:${GIT_TOKEN}@github.com/${GITHUB_REPOSITORY}.git"
          git fetch origin main
```

**Comment by samuelfernandez** [comment]
All the points you mention were on my mind, thanks for the detailed report. I'll create issues and follow-ups and keep you in the loop

**Comment by samuelfernandez** [comment]
@wSedlacek I cc'ed you in two follow up issues to address the points you mention. I hope they are all covered.

Regarding the issue with git, here you have some docs: https://github.com/ng-easy/platform/blob/main/libs/builders/README.md#tips-for-github-actions-ci

The important part is this:

```yml
      - name: Checkout repo
        uses: actions/checkout@v2.3.4
        with:
          fetch-depth: 0
          persist-credentials: false # Needed so that semantic release can use the admin token
``` 

Hope it helps! 😄

**Comment by wSedlacek** [comment]
Yeah when I set persist creds to false it did prevent the creds from the checkout from leaking, but I wasn't able to do the fetch afterwards nor the push later on.

I am using a private repo and have my main branch protected so I need git configured with creds that have admin rights to the repo. Hence the manual configuration with the remote.

**Pull request merged**